CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ option(TRANSPORT_PAHO "Build IoT Samples with Paho MQTT support" OFF)
 option(PRECONDITIONS "Build SDK with preconditions enabled" ON)
 option(LOGGING "Build SDK with logging support" ON)
 option(ADDRESS_SANITIZER "Build with address sanitizer" OFF)
+set(AZ_MQTT5_USER_CONFIG_PATH "" CACHE STRING "Path to folder containing user configuration for MQTT5 (az_mqtt5_user_config.h). If not set, default configuration will be used.")
```

Comment:
So what's going to happen if another component needs to get a user config?
Will we add another option for setting a separate configuration path for that component?
Wouldn't it be better to have a single option for a single path and taking a generic config header for the entire SDK? Or at least a single config path, and keeping the headers separate.

Reply:
This should really be verified by the Azure SDK team.

Reply:
That's a good point, given that we are moving these mqtt5 files outside of core, the mqtt5 config file will be specific to that new folder, not core. I don't foresee any configuration changes for the core folders but we can address with the SDKs team.